libvast/builtins/connectors/file.cpp

```diff
@@ -0,0 +1,203 @@
+#include <vast/detail/fdinbuf.hpp>
+#include <vast/detail/file_path_to_parser.hpp>
+#include <vast/detail/posix.hpp>
+#include <vast/detail/string.hpp>
+#include <vast/logger.hpp>
+#include <vast/plugin.hpp>
+
+#include <caf/error.hpp>
+
+#include <fcntl.h>
+#include <filesystem>
+#include <memory>
+#include <string_view>
+#include <unistd.h>
+
+namespace vast::plugins::file {
+
+class plugin : public virtual loader_plugin {
+public:
+  static constexpr auto max_chunk_size = size_t{16384};
+  static constexpr auto stdin_path = "-";
+
+  auto
+  make_loader(std::span<std::string const> args, operator_control_plane&) const
+    -> caf::expected<generator<chunk_ptr>> override {
+    auto read_timeout = read_timeout_;
+    auto path = std::string{};
+    auto following = false;
+    auto is_socket = false;
+    for (auto i = size_t{0}; i < args.size(); ++i) {
+      const auto& arg = args[i];
+      VAST_TRACE("processing loader argument {}", arg);
+      if (arg == "--timeout") {
+        if (i + 1 == args.size()) {
+          return caf::make_error(ec::syntax_error,
+                                 fmt::format("missing duration value"));
+        }
+        if (auto parsed_duration = to<vast::duration>(args[i + 1])) {
+          read_timeout = std::chrono::duration_cast<std::chrono::milliseconds>(
+            *parsed_duration);
+          ++i;
+        } else {
+          return caf::make_error(ec::syntax_error,
+                                 fmt::format("could not parse duration: {}",
+                                             args[i + 1]));
+        }
+      } else if (arg == "-" || arg == "stdin") {
+        path = stdin_path;
+      } else if (arg == "-f" || arg == "--follow") {
+        following = true;
+      } else if (not arg.starts_with("-")) {
+        std::error_code err{};
+        auto status = std::filesystem::status(arg, err);
+        if (err) {
+          return caf::make_error(ec::parse_error,
+                                 fmt::format("could not access file {}: {}",
+                                             arg, err));
+        }
+        is_socket = (status.type() == std::filesystem::file_type::socket);
+        path = arg;
+      } else {
+        return caf::make_error(
+          ec::invalid_argument,
+          fmt::format("unexpected argument for 'file' connector: {}", arg));
+      }
+    }
+    if (path.empty()) {
+      return caf::make_error(ec::syntax_error,
+                             fmt::format("no file specified"));
+    }
+    auto fd = STDIN_FILENO;
+    if (is_socket) {
+      if (path == stdin_path) {
+        return caf::make_error(ec::filesystem_error, "cannot use STDIN as UNIX "
+                                                     "domain socket");
+      }
+      auto uds = detail::unix_domain_socket::connect(path);
+      if (!uds) {
+        return caf::make_error(
+          ec::filesystem_error,
+          fmt::format("Unable to connect to UNIX domain socket at {}:", path));
```

Comment:
```suggestion
          fmt::format("unable to connect to UNIX domain socket at {}", path));
```

Reply:
Lower-cased the error message in https://github.com/tenzir/vast/pull/3088/commits/26c17b32e7ba53a27f5027a6ab918f74b40d215c

Reply:
You missed the trailing `:` 😄

Reply:
Not anymore :sweat_smile: